packages/flutter_tools/templates/app/lib/main.dart.tmpl

```diff
@@ -38,111 +38,182 @@ void main() {
 }
 
 {{^withPluginHook}}
+// This is a stateless widget that is the root widget for this application.
+//
+// It is said to be "stateless" because its attributes are fixed at creation, so
+// it can't keep any modified application state itself. The state it uses all
+// comes from its inputs. The power of stateless widgets is that although they
+// are lightweight and immutable, they can have logic in their build methods to
+// modify what gets built based on their configuration.
 class MyApp extends StatelessWidget {
   const MyApp({super.key});
 
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Flutter Demo',
+      title: 'Counter Example',
       theme: ThemeData(
-        // This is the theme of your application.
+        // This is the theme of the application.
+        //
+        // TRY THIS: Try running the application with "flutter run" on the
+        // command line, or the run button in the IDE. You'll see the
+        // application has a purple toolbar. Then, without quitting the app, try
+        // changing the colorScheme below to use a seedColor of Colors.green and
+        // then invoke "hot reload" (press "r" in the console where you ran
+        // "flutter run", save the changes to "hot reload" in a Flutter IDE, or
+        // press the "hot reload" button in a Flutter IDE).
```

Comment:
this parenthetical got more complicated in a confusing way. can we drop the added clause?

Reply:
We can.  I added it because not all users have "reload on save" set.  It is the default, though.

Reply:
I think in general the template needs to be a nice and short and quick introduction to the basics so that someone can orient themselves. It's not a tutorial or a codelab, and it's definitely not formal api documentation. There's a huge risk of overwhelming a new reader with too much information.